caom2-artifact-resolvers/src/main/java/ca/nrc/cadc/caom2/artifact/resolvers/XmmResolver.java

```diff
@@ -0,0 +1,75 @@
+/*
+ ************************************************************************
+ ****  C A N A D I A N   A S T R O N O M Y   D A T A   C E N T R E  *****
+ *
+ * (c) 2017.                            (c) 2017.
+ * National Research Council            Conseil national de recherches
+ * Ottawa, Canada, K1A 0R6              Ottawa, Canada, K1A 0R6
+ * All rights reserved                  Tous droits reserves
+ *
+ * NRC disclaims any warranties         Le CNRC denie toute garantie
+ * expressed, implied, or statu-        enoncee, implicite ou legale,
+ * tory, of any kind with respect       de quelque nature que se soit,
+ * to the software, including           concernant le logiciel, y com-
+ * without limitation any war-          pris sans restriction toute
+ * ranty of merchantability or          garantie de valeur marchande
+ * fitness for a particular pur-        ou de pertinence pour un usage
+ * pose.  NRC shall not be liable       particulier.  Le CNRC ne
+ * in any event for any damages,        pourra en aucun cas etre tenu
+ * whether direct or indirect,          responsable de tout dommage,
+ * special or general, consequen-       direct ou indirect, particul-
+ * tial or incidental, arising          ier ou general, accessoire ou
+ * from the use of the software.        fortuit, resultant de l'utili-
+ *                                      sation du logiciel.
+ *
+ ****  C A N A D I A N   A S T R O N O M Y   D A T A   C E N T R E  *****
+ ************************************************************************
+ */
+
+package ca.nrc.cadc.caom2.artifact.resolvers;
+
+import ca.nrc.cadc.caom2.artifact.resolvers.util.ResolverUtil;
+import ca.nrc.cadc.net.StorageResolver;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import org.apache.log4j.Logger;
+
+/**
+ * This class can convert a XMM URI into a URL.
+ *
+ * @author jeevesh
+ */
+public class XmmResolver implements StorageResolver {
+    public static final String SCHEME = "xmm";
+    private static final Logger log = Logger.getLogger(XmmResolver.class);
+    private static final String BASE_ARTIFACT_URL = "http://nxsa.esac.esa.int/nxsa-sl/servlet/data-action-aio?";
+    private static final String CANNOT_GET_URL = "Can't generate URL from URI.";
+
+
+    public XmmResolver() {
+    }
+
+    @Override
+    public String getScheme() {
+        return SCHEME;
+    }
+
+    @Override
+    public URL toURL(URI uri) {
```

Comment:
As previously noted, if we really need a stub (??) the  toURL should throw UnsupportedOperationException